cmd/compose/port.go

```diff
@@ -62,19 +68,45 @@ func portCommand(p *ProjectOptions, dockerCli command.Cli, backend api.Service)
 	return cmd
 }
 
+func printPorts(dockerCli command.Cli, publishers api.PortPublishers, opts portOptions) {
+	for _, publisher := range publishers {
+		if publisher.Protocol != opts.protocol {
+			continue
+		}
+
+		var format string
+		if strings.Contains(publisher.URL, ":") {
+			format = "%d/%s -> [%s]:%d\n"
+		} else {
+			format = "%d/%s -> %s:%d\n"
+		}
+		fmt.Fprintf(dockerCli.Out(), format, publisher.PublishedPort, publisher.Protocol, publisher.URL, publisher.TargetPort)
+	}
+}
+
 func runPort(ctx context.Context, dockerCli command.Cli, backend api.Service, opts portOptions, service string) error {
 	projectName, err := opts.toProjectName(ctx, dockerCli)
 	if err != nil {
 		return err
 	}
-	ip, port, err := backend.Port(ctx, projectName, service, opts.port, api.PortOptions{
-		Protocol: opts.protocol,
-		Index:    opts.index,
-	})
-	if err != nil {
-		return err
-	}
 
-	fmt.Fprintf(dockerCli.Out(), "%s:%d\n", ip, port)
-	return nil
+	if opts.list {
+		summaries, err := backend.Ps(ctx, projectName, api.PsOptions{Services: []string{service}})
+		if err != nil {
+			return err
+		}
+		printPorts(dockerCli, summaries[opts.index].Publishers, opts)
```

Comment:
We need to test the size of the `summaries` slice otherwise we would potentially trigger `slice bounds out of range` error